core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -434,6 +435,61 @@ public boolean schemaExists(Session session, CatalogSchemaName schema)
         return Optional.empty();
     }
 
+    @Override
+    public Optional<TableExecuteHandle> getTableHandleForExecute(Session session, TableHandle tableHandle, String procedure, Map<String, Object> executeProperties, Constraint constraint)
```

Comment:
Can we envision procedures for anything other than tables?

For example, i could have a procedure to convert the table format (eg change from CSV to Parquet). Could i apply it to materialized view too?
then we would need not to have `TableHandle` here.

Reply:
It makes sense. But I do not see an obvious way to generalize SPI to handle that. I would suggest to keep this PR focused and broden the scope as a followup when we need that. It may result in some backward incompatible changes to the SPI later on, but it should not be a big deal. We are doing that from time to time.